### PR TITLE
bootctl: ignore invalid boot entries

### LINKED ISSUE #25359: With systemd 252.1-1 one machine has an error output from bootctl status
### systemd version the issue has been seen with

systemd 252.1-1

### Used distribution

Arch Linux

### Linux kernel version used

6.0.8

### CPU architectures issue was seen on

_No response_

### Component

bootctl

### Expected behaviour you didn't see

I expected no error lines starting Failed to read boot option, when running bootctl status.

### Unexpected behaviour you saw

$ sudo bootctl status
Failed to read boot option 8194: No such file or directory
Failed to read boot option 8196: No such file or directory
System:
      Firmware: UEFI 2.70 (INSYDE Corp. 3865.00)
 Firmware Arch: x64
   Secure Boot: disabled (disabled)
  TPM2 Support: yes
  Boot into FW: supported

### Steps to reproduce the problem

Run bootctl status

### Additional program output to the terminal or log subsystem illustrating the issue

_No response_

### PR DESCRIPTION
**Pull request by yuwata**
Fixes #25359.

### CODE REVIEW

File: src/boot/bootctl.c [hunk 1]
```diff
@@ -589,10 +589,9 @@ static int print_efi_option(uint16_t id, int *n_printed, bool in_order) {
                 return log_debug_errno(r, "Failed to read boot option 0x%04X: %m", id);
 
         /* print only configured entries with partition information */
-        if (!path || sd_id128_is_null(partition)) {
-                log_debug("Ignoring boot entry 0x%04X without partition information.", id);
-                return 0;
```
Comment (bluca) [inline src/boot/bootctl.c:594]: a future use might not do the same, so better to keep returning 0 here to avoid future issues

Comment (yuwata) [reply src/boot/bootctl.c:594]: This is not a library function, but only used in a specific place of bootctl. Hence it is not necessary to consider any future usages.

But, OK, this change is not important for this PR. I will drop it.

File: src/boot/bootctl.c [hunk 2]
```diff
@@ -586,7 +586,7 @@ static int print_efi_option(uint16_t id, int *n_printed, bool in_order) {
 
         r = efi_get_boot_option(id, &title, &partition, &path, &active);
         if (r < 0)
-                return log_error_errno(r, "Failed to read boot option 0x%04X: %m", id);
+                return log_debug_errno(r, "Failed to read boot option 0x%04X: %m", id);
```
Comment (poettering) [inline src/boot/bootctl.c:589]: hmm, this should not propagate the error then I guess. i.e. return 0. Also say `, ignoring` as usual

Comment (poettering) [reply src/boot/bootctl.c:589]: → #25439

**Pull request merged**